sdk/go/common/resource/config/crypt.go

```diff
@@ -31,14 +32,19 @@ import (
 // Encrypter encrypts plaintext into its encrypted ciphertext.
 type Encrypter interface {
 	EncryptValue(plaintext string) (string, error)
+	EncryptValueWithContext(ctx context.Context, plaintext string) (string, error)
```

Comment:
This is a breaking change for code that implements the interface but not code that consumes it.

Reply:
If we want to be backward compatible, I think we need a new interface. It's not an uncommon pattern to add a `EncrypterWithContext` interface that requires `Encrypter` and then try down casting from `Encrypter` to `EncrypterWithContext` when you need to specify the context. This is how the `error` module does error chaining.

Reply:
I was fearing that as it seems to get baroque quickly, but I'll try to work it out. The benefit in the PR is not worth imposing a breaking change.

Reply:
I'm not sure why this stuff is even in common. It's probably fine to just change it.

Reply:
So 
A) This is already a breaking change because you've added methods to an interface
B) I don't think we care about breaking changes in common unless they're likely used by programs, so I'd rather take the "simpler" breaking change of just adding context to the existing Encrypt/Decrypt methods.

Reply:
That sounds good to me! If we're pretty sure user code does not call Encrypter. I will edit and add a changelog.